Sources/WebSocketKit/WebSocket.swift

```diff
@@ -192,6 +241,41 @@ public final class WebSocket {
         }
     }
 
+    private func pingAndScheduleNextTimeoutTask() {
+        guard channel.isActive, let timeout = timeout else {
+            return
+        }
+
+        self.sendPing()
+
+        let timeSinceLastRead = .now() - lastReadTime
+        if timeSinceLastRead >= timeout {
```

Comment:
If we send a ping, I think we should ensure we get a pong back specifically. Right now the logic only checks to see if _something_ has been read since the last ping.